### PR TITLE
Add elasticsearch config

### PR DESCRIPTION
**Pull request by shubham-padia**
partially #3467
single node version of https://github.com/pires/kubernetes-elasticsearch-cluster

### CODE REVIEW

File: kubernetes/yamls/elasticsearch/es-rc.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        - name: "DISCOVERY_SERVICE"
+          value: "elasticsearch"
+        - name: NODE_MASTER
+          value: "true"
+        - name: NODE_DATA
+          value: "true"
+        - name: HTTP_ENABLE
+          value: "true"
+        - name: "ES_JAVA_OPTS"
+          value: "-Xms256m -Xmx256m"
```
Comment (shubham-padia) [inline kubernetes/yamls/elasticsearch/es-rc.yaml]: > By default, ES_JAVA_OPTS is set to -Xms256m -Xmx256m. This is a very low value but many users, i.e. minikube users, were having issues with pods getting killed because hosts were out of memory. One can change this in the deployment descriptors available in this repository.

File: kubernetes/yamls/elasticsearch/es-rc.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  volumeClaimTemplates:
+  - metadata:
+      name: storage
+      annotations:
+        volume.beta.kubernetes.io/storage-class: standard
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 12Gi
```
Comment (shubham-padia) [inline kubernetes/yamls/elasticsearch/es-rc.yaml]: @niranjan94 Please review this once

**Comment by shubham-padia** [comment]
@niranjan94 Please review

File: kubernetes/yamls/web/ingress-notls.yml [hunk 3]
```diff
@@ -20,3 +20,7 @@ spec:
         backend:
           serviceName: redirector
           servicePort: 8080
+      - path: /
```
Comment (niranjan94) [inline kubernetes/yamls/web/ingress-notls.yml]: No ingress needed. ES will be accessed locally.

File: kubernetes/yamls/web/ingress-tls.yml [hunk 4]
```diff
@@ -26,3 +26,7 @@ spec:
         backend:
           serviceName: web
           servicePort: 8080
+      - path: /
```
Comment (niranjan94) [inline kubernetes/yamls/web/ingress-tls.yml]: No ingress needed. ES will be accessed locally.

File: requirements/common.txt [hunk 5]
```diff
@@ -52,3 +52,4 @@ python-geoip-geolite2
 pycountry
 pytz
 diff-match-patch
+google-compute-engine
```
Comment (niranjan94) [inline requirements/common.txt]: Why is this needed ? We don't want to have any code dependency on google compute engine

File: kubernetes/yamls/elasticsearch/es-rc.yaml [hunk 6]
```diff
@@ -0,0 +1,67 @@
+apiVersion: v1
+kind: ReplicationController
```
Comment (niranjan94) [inline kubernetes/yamls/elasticsearch/es-rc.yaml]: Instead of a replicationcontroller, use a deployment. 

Ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

File: kubernetes/yamls/elasticsearch/es-svc.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: elasticsearch
+  labels:
+    component: elasticsearch
+spec:
+  type: LoadBalancer
```
Comment (niranjan94) [inline kubernetes/yamls/elasticsearch/es-svc.yaml]: Remove this. Not needed. This will be locally accessible

File: kubernetes/yamls/elasticsearch/es-svc.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
```
Comment (niranjan94) [inline kubernetes/yamls/elasticsearch/es-svc.yaml]: Rename it as `elasticsearch-service.yml`

File: kubernetes/yamls/elasticsearch/es-rc.yaml [hunk 9]
```diff
@@ -0,0 +1,67 @@
+apiVersion: v1
```
Comment (niranjan94) [inline kubernetes/yamls/elasticsearch/es-rc.yaml]: Rename as `elasticsearch-deployment.yml`

File: kubernetes/yamls/elasticsearch/es-rc.yaml [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        - containerPort: 9200
+          name: http
+          protocol: TCP
+        - containerPort: 9300
+          name: transport
+          protocol: TCP
+        volumeMounts:
+        - mountPath: /data
+          name: storage
+  volumeClaimTemplates:
```
Comment (niranjan94) [inline kubernetes/yamls/elasticsearch/es-rc.yaml]: Instead of this, add a persistent disk similar to how we are doing in the postgres config.

File: requirements/common.txt (hunk 5 above)
Comment (shubham-padia) [inline requirements/common.txt]: This is not related to the PR, travis fails without this. Remember the switch to the new travis images sometime ago in nextgen ?

File: kubernetes/yamls/web/ingress-tls.yml (hunk 4 above)
Comment (shubham-padia) [inline kubernetes/yamls/web/ingress-tls.yml]: @niranjan94 How will the clients be making a query then ?

File: kubernetes/yamls/elasticsearch/es-rc.yaml (hunk 10 above)
Comment (shubham-padia) [inline kubernetes/yamls/elasticsearch/es-rc.yaml]: How much memory ?

**Comment by shubham-padia** [comment]
@niranjan94 made the requested changes as I deemed fit, please review again :)
